### DISCUSSION TITLE
Kubernetes Ingress Nginx configuration for Websocket support

### DISCUSSION THREAD

**Discussion by niklasfrick** [dani-garcia/vaultwarden#5144] [Ideas]
I'd like to share my Helm values for configuring Ingress Nginx to support WebSockets. This configuration has been working well for me, and I thought it might be helpful for others.

First, it's important to note that `allowSnippetAnnotations` must be set to `true` in the Ingress Nginx config for this to work.
Here are the annotations I'm using in my Helm values for the ingress (make sure to adjust it to your setup/helm chart if necessary):

```
ingress:
  additionalAnnotations:
    nginx.ingress.kubernetes.io/configuration-snippet: |
      proxy_set_header Upgrade "websocket";
      proxy_set_header Connection "Upgrade";
```

**Comment by mhutter** [comment]
I'm a bit confused as to why this is required; according to the [upstream Ingress-nginx docs](https://kubernetes.github.io/ingress-nginx/user-guide/miscellaneous/#websockets) websockets should work out of the box, but in my experience the config snippet (which in turn requires enabling the `allowSnippetAnnotations` setting) is required.

**Reply by BlackDex** [reply]
I think it depends on the type.
If the type is HTTP instead of TCP, it might have issues?
Also, it needs to talk HTTP/1.1 to Vaultwarden, anything else will cause issues, since HTTP/1.0 and HTTP/2.0 do not support websocket connections.
